src/simstation/Simulation.java

```diff
@@ -33,10 +32,13 @@ public void populate(){
 //    }
 //
     public void start() {
+        if (world.isEmpty()) {
```

Comment:
Start it is creating a new set of Agents and start them. This is not resuming, so you should call populate even when r world not empty (since populate will initialize that anyways)